.github/workflows/wheel-builder.yml

```diff
@@ -30,58 +33,71 @@ jobs:
           path: bcrypt-wheelhouse/
 
   macos:
     runs-on: macos-latest
     strategy:
       matrix:
         PYTHON:
-          - {VERSION: '3.8', ABI_VERSION: 'cp36'}
+          - ABI_VERSION: 'cp36'
+            DOWNLOAD_URL: 'https://www.python.org/ftp/python/3.8.5/python-3.8.5-macosx10.9.pkg'
+            BIN_PATH: '/Library/Frameworks/Python.framework/Versions/3.8/bin/python3'
     name: "${{ matrix.PYTHON.ABI_VERSION }} macOS"
     steps:
       - uses: actions/checkout@master
-      - name: Setup python
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.PYTHON.VERSION }}
-
-      - run: pip install -U wheel cffi six
-      - run: pip download bcrypt==${{ github.event.client_payload.BUILD_VERSION }} --no-deps --no-binary bcrypt && tar zxvf bcrypt*.tar.gz && mkdir wheelhouse
-      - run: cd bcrypt* && python setup.py bdist_wheel --py-limited-api=${{ matrix.PYTHON.ABI_VERSION }} && mv dist/bcrypt*.whl ../wheelhouse
-      - run: pip install -f wheelhouse --no-index bcrypt
       - run: |
-          python -c "import bcrypt;password = b'super secret password';hashed = bcrypt.hashpw(password, bcrypt.gensalt());bcrypt.checkpw(password, hashed)"
+          curl "${{ matrix.PYTHON.DOWNLOAD_URL }}" -o python.pkg
+          sudo installer -pkg python.pkg -target /
+      - run: ${{ matrix.PYTHON.BIN_PATH }} -m venv venv
+      - run: venv/bin/pip install -U pip wheel cffi six
+      - run: venv/bin/pip download bcrypt==${{ github.event.inputs.version }} --no-deps --no-binary bcrypt && tar zxvf bcrypt*.tar.gz && mkdir wheelhouse
+      - run: cd bcrypt* && ../venv/bin/python setup.py bdist_wheel --py-limited-api=${{ matrix.PYTHON.ABI_VERSION }} && mv dist/bcrypt*.whl ../wheelhouse
+      - run: venv/bin/pip install -f wheelhouse --no-index bcrypt
+      - run: |
+          venv/bin/python -c "import bcrypt;password = b'super secret password';hashed = bcrypt.hashpw(password, bcrypt.gensalt());bcrypt.checkpw(password, hashed)"
 
       - run: mkdir bcrypt-wheelhouse
       - run: mv wheelhouse/bcrypt*.whl bcrypt-wheelhouse/
       - uses: actions/upload-artifact@v1
         with:
-          name: "bcrypt-${{ github.event.client_payload.BUILD_VERSION }}-macOS-${{ matrix.PYTHON.ABI_VERSION }}"
+          name: "bcrypt-${{ github.event.inputs.version }}-macOS-${{ matrix.PYTHON.ABI_VERSION }}"
           path: bcrypt-wheelhouse/
 
   windows:
     runs-on: windows-latest
     strategy:
       matrix:
         WINDOWS:
-          - {ARCH: 'x86', WINDOWS: 'win32'}
-          - {ARCH: 'x64', WINDOWS: 'win64'}
+          - 'x86'
+          - 'x64'
         PYTHON:
-          - {VERSION: "3.6", ABI_VERSION: "cp36"}
-    name: "${{ matrix.PYTHON.ABI_VERSION }} ${{ matrix.WINDOWS.WINDOWS }}"
+          - {VERSION: "3.6", USE_ABI3: "True", ABI_VERSION: "cp36"}
+          # These jobs are temporary until we feel comfortable shipping wheels that
```

Comment:
Counter point: Let's force the matter. upgrading pip is trivial.

Reply:
The way this will manifest is a super opaque error to the user that it can’t find a compiler, so I definitely disagree. If it just said no suitable installation candidate that’d be fine, but an error sending them down the wrong path is a really bad UX.

Reply:
This will be my last counter argument: we're incredibly conservative in using new stuff for pyca projects. I think too conservative. We're constantly maintaing support for older things, and not using modern packaging features. I think this is a contributing factor to why we both find these projects so frustrating to work on. We ought to be able to do something as simple as get people to upgrade their pips.